modules/shortcodes/soundcloud.php

```diff
@@ -58,7 +58,7 @@ function soundcloud_shortcode( $atts, $content = null ) {
 
 	$player_type = soundcloud_get_option( 'player_type', 'visual' );
 	$isIframe    = $player_type !== 'flash';
-	$isVisual    = ! $player_type || $player_type === 'visual' || $shortcode_options['visual'];
+	$isVisual    = ! $isIframe || $player_type === 'visual' || ( isset( $shortcode_options['visual'] ) && false != $shortcode_options['visual'] );
 
```

Comment:
The previous logic didn't appear to work at all in my tests. The SoundCloud widget always used the Visual appearance, regardless of your settings.

I'd be happy to rework on that logic completely though, to make it more readable maybe?